### DISCUSSION TITLE
Renovate app unable to connect to private Docker registry

### DISCUSSION THREAD

**Discussion by tn185070** [renovatebot/renovate#18167] [Request Help] [answered]
Hi, I'm trying to configure the Renovate GitHub app to connect to my organization private Artifactory Docker registry. It is connecting successfully to our Artifactory Maven registry but for some reason I can't get it to connect to the Docker registry properly.

I've tried a few different configs but the current one is like:

```
"hostRules": [
    {
      "matchHost": "https://my-maven-artifactory",
      "hostType": "maven",
      "username": "xxx",
      "encrypted": {
        "password": "wcFMA/xxx"
      }
    },
    {
      "matchHost": "https://my-artifactory-docker-registry",
      "hostType": "docker",
      "authType": "basic",
      "encrypted": {
        "token": "wcFMA/yyy"
      }
    }
  ],
  "packageRules": [
    {
      "matchPackagePrefixes": ["my.org"],
      "registryUrls": ["https://my-maven-artifactory"]
    },
    {
      "matchManagers": ["dockerfile"],
      "registryUrls": ["https://my-artifactory-docker-registry"]
    }
```

Here are the relevant logs:

```
DEBUG: Datasource 404
{
  "datasource": "docker",
  "packageName": "my-artifactory-docker-registry/my-image",
  "url": "https://my-artifactory-docker-registry/v2/library/library/my-image/tags/list?n=10000"
}
DEBUG: Failed to look up dependency my-artifactory-docker-registry/my-image (my-artifactory-docker-registry/my-image)(packageFile="distribution/container/Dockerfile", dependency="my-artifactory-docker-registry/my-image")
...
DEBUG: http statistics
{
  "urls": {
    ...
    "https://my-artifactory-docker-registry/artifactory/api/docker/artifactory-docker-registry/v2/token (GET,200)": 3,
    "https://my-artifactory-docker-registry/v2/my-image/tags/list (GET,401)": 1,
    "https://my-artifactory-docker-registry/v2/my-image/tags/list (GET,404)": 1,
    "https://my-artifactory-docker-registry/v2/library/my-image/tags/list (GET,401)": 1,
    "https://my-artifactory-docker-registry/v2/library/my-image/tags/list (GET,404)": 1,
    "https://my-artifactory-docker-registry/v2/library/library/my-image/tags/list (GET,401)": 1,
    "https://my-artifactory-docker-registry/v2/library/library/my-image/tags/list (GET,404)": 1,
```

If I call `https://my-artifactory-docker-registry/v2/my-image/tags/list` from Postman with the token provided in the configuration, I am able to get a a list of the image tags. I'm confused why Renovate is making a call to get a token from Artifactory, when I have already provided the token in the configuration. I suspect the token from this request is being used in subsequent calls to Artifactory and is causing the issue. I tried getting a token from `https://my-artifactory-docker-registry/artifactory/api/docker/artifactory-docker-registry/v2/token` myself and found that it didn't work when querying Artifactory from Postman. 

BTW the token in the configuration is just a base64'd username:password, I tried that since I was hoping that proving Renovate with a token would prevent it from trying to get one from Artifactory (previously I just put the username and password directly in the configuration).

Would appreciate any insight into this.

**Comment by viceice** [comment, marked as answer]
use username and password for docker instead of token and auth type

**Reply by tn185070** [reply]
Sigh. We tried that earlier but it wasn't working. Sometime after I switched to using a token I must have made another change that resolved the initial issue. Thanks for your prompt response. I'll try to pay it forward.
